lib/classes/Variables.js

```diff
@@ -40,7 +40,7 @@ class Variables {
 
     _.deepMapValues(this.service, (property, propertyPath) => {
       if (typeof property === 'string') {
-        const populateSingleProperty = new Promise((resolve) => this
+        const populateSingleProperty = new BbPromise((resolve) => this
```

Comment:
Let me try to decrypt this:
```js
const populateSingleProperty = this.populateProperty(property, true)
  .then(newProperty => _.set(this.service, propertyPath, newProperty)
  .return();
```

This should be equal and seems more straight forward to me. `new BbPromise()` should really be avoided and isn't even necessary here. If any function doesn't return a BbPromise but a plain native Promise, it's easiest to wrap it using `BbPromise.resolve(myNativePromiseFunc()).then(...)`.